## On cellular control networks

Back in graduate school, I became interested in systems biology, but I never quite found what I was looking for. Since then I've stumbled through papers on [gene-set enrichment analysis](http://amp.pharm.mssm.edu/Enrichr/), [grouped association tests](https://www.sciencedirect.com/science/article/pii/S0002929718301083), [critical state transitions](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5189937/), [stochastic](https://www.ncbi.nlm.nih.gov/pubmed/17037977) [modeling](https://www.ncbi.nlm.nih.gov/pubmed/28341132) and [related inference methods](https://darrenjw.wordpress.com/), large-scale [perturbation](https://www.sciencedirect.com/science/article/pii/S0092867416316105) [experiments](https://www.biorxiv.org/content/early/2017/05/10/136168) or [correlation](https://www.ncbi.nlm.nih.gov/pubmed/15867157) [networks](https://www.ncbi.nlm.nih.gov/pubmed/29202695), none of which quite expressed what I was looking for. Some of these enable highly detailed modeling motivated by both the biology and the underlying physics. Unfortunately, these do not scale well; they tend to include fewer than 20 interacting genes and proteins, whereas modern sequencing datasets routinely measure $O(10^4)$ genes. Others grapple head-on with genome-wide inference, but do not explicitly model direct interactions between transcriptional regulators and their targets. I want to see inference of targets and regulators, genome-wide, integrating as much data as necessary to improve performance. I also want to see people thoughtfully consider [the limits of](https://www.ncbi.nlm.nih.gov/pubmed/10380182) [what is possible](https://reader.elsevier.com/reader/sd/pii/S0304407616301828?token=7DC7B7BC54B12D6F3DC4EC94B02DE7C119374F7EB120842D3BA61CE7722169116D344AE963EEB8879FCD391DCB145C4E), how to [compactly represent the uncertainty remaining](https://public.econ.duke.edu/~kdh9/Source%20Materials/Research/4.%20Searching%20for%20the%20Causal%20Structure%20of%20a%20VAR.pdf) after all evidence has been exhausted, and how to use resources optimally for gene regulatory inference.

Some time over the last three years, I managed to pin down my questions, and I began finding some of the right papers to satisfy my curiosity. For better or for worse, there are tons. In this series on systems biology, I will try to tackle them in small, coherent groups. 

- A few data resources
- ChIP-seq, ATAC-seq, enhancers, and motif analysis
- Three methods of pairing enhancers with their target genes
- Identifiability of gene regulatory networks
- Inferring gene regulatory networks from RNA data part 1: the DREAM5 competition
- Inferring gene regulatory networks from RNA data part 2: miscellaneous single-cell RNA-seq paper I've run across
- Inferring gene regulatory networks from combined RNA and DNA sequencing data
