---
layout: post
title: Learning to learn about epigenetics (project description)
math: true
tags: misc
---

This is a description of the research project from [my class on epigenetics](https://ekernf01.github.io/heart-epigenetics).


---

## Final project 

### Description

The majority of the course sessions will be devoted to an individual research project. You will use this to practice fundamental research skills. You don't need to "finish" it. You just need to show active progress. 

The main output of this project is an **annotated bibliography** that describes the premise of the project, the findings, and the sources each part is based on. It should be arranged and written in a way that is intuitive to someone encountering your questions for the first time. To produce this document, follow these steps. 

**Explore.**

Choose a subfield of epigenetics and do some exploratory background reading. Document the sources that were useful to you in learning the overall shape of the field and the major new things you learned. Put this in a preface to your annotated bibliography. 

**Choose a question.**

Choose a narrow question to focus on exclusively. Many suggestions are provided below. You may also propose your own ideas. 

**Root out the original research underlying the premise of the question.**

Your question will probably rest on some key claims. For example, if you want to know whether enhancer RNAs are functionally important, your question implies that enhancers are indeed transcribed into RNAs. How is that known and when was it first discovered? If the key claim says that histone acetyltransferases promote transcription by chemically modifying the tail of the H3 histone protein, who first learned that these chemical modifications occur in H3? What did they physically do in order to learn that? Who first identified a histone acetyltransferase enzyme? What did they physically do in order to learn that? 

Depending how far your project is from the basic building blocks, you may not have time to dig up every single building block. The objective is to get some practice doing this, so it does not need to be exhaustive. You can assume certain things from a textbook or review article and focus on the primary research behind others.

**Answer the question.**

Find papers that address the core question and add their most important findings to your document. For example, if you want to know whether enhancer RNA's are functionally important, find papers where the authors knocked down an enhancer RNA and checked to see if it still helped activate transcription at its target gene. If you want to know whether histone acetyltransferases activate transcription, then look for papers where people disabled histone acetyletransferases and then measured transcription.

**Reflect on the future.**

Is the state of the science completely solved in the area you chose? Do we know everything there is to know? If so, start over at step 1, and choose a new project. If not, what are you still curious about in this area? Has anyone investigated it, or are there opportunities to do so? Extend your annotated bibliography with a "future outlook" section talking about what the field should ask next. If you're not yet at the cutting edge of the field, talk about what you would want to read next. 


### The chalk talk

Aside from reading and writing, another important skill is communicating this type of work to your colleagues. You have a chance to practice this midway through the project, in an informal "chalk talk". Each student will sign up for a 10 minute slot, with a few minutes to talk and show diagrams and a few more minutes to brainstorm next steps with colleagues. Depending on how this goes, we may do it more than once. The objective of the chalk talk is to convey the premise of the project, the question, and one important paper or piece of evidence about the premise or the answer. You should make about 1 slide, showing any image or piece of data that is essential for you to tell the story.

### Topics

Here are some possible topics for you to investigate. 

- Olfactory receptors. The human genome contains over 1000 olfactory receptor genes. Each olfactory neuron activates exactly one of them. Why is this necessary? How does it happen? A good place to start might be:

    > Lyons DB, Allen WE, Goh T, Tsai L, Barnea G, Lomvardas S. An epigenetic trap stabilizes singular olfactory receptor expression.Cell. 2013 Jul 18;154(2):325-36. doi: 10.1016/j.cell.2013.06.039.

- X chromosome dosage. In most animals, if an embryo has an extra or deleted chromosome, it will die. One remarkable exception is the mammalian X chromosome: females have two, males have one. Mammalian XX cells inactivate one X chromosome to compensate. How and when does this happen? 

    X inactivation is complex and it has been studied for a long time, and you will probably find that even answering "how does it work and how do we know" is too big for this project. You will need to devote extra energy to choosing a smaller part of this question to answer (I'll help). Here is a useful starting point.

    > Fang, H., Disteche, C. M., & Berletch, J. B. (2019). X inactivation and escape: epigenetic and structural features. Frontiers in Cell and Developmental Biology, 7, 219.
    

- Inherited, acquired, non-genetic traits (Sweden). Detailed records on diet and health outcomes from a study in the northern part of Sweden (Överkalix) showed strange, sex-dependent patterns of inheritance. What exactly did they find? Why are these phenomena regarded as not genetic but rather epigenetic? What might be the mechanisms? 

    > https://en.wikipedia.org/wiki/%C3%96verkalix_study

- Inherited, acquired, non-genetic traits (Netherlands). In the late stages of the second world war (1944-45), the western Netherlands experienced a horrific wintertime famine. Considering women pregnant during the famine, scientific studies found that their children were affected, but also, their *grandchildren* were affected. What was one disease or effect that propagated in this way, and how was it documented? Why was this type of inheritance considered to be not genetic but epigenetic? What might the mechanisms be?

    > https://en.wikipedia.org/wiki/Dutch_famine_of_1944%E2%80%931945#Legacy

- DNA methylation. DNA methylation is a fundamental component of gene regulation. It helps silence viruses that have integrated into the genome, and it also is an important factor controlling gene activity in cancer and during development. There are many basic questions about DNA methylation that you could answer in the course of a project. For example: how do we know that methylated DNA exists? How do we know that it represses nearby gene activity? Does it repress transcription by directly preventing polymerases from binding, or through some more complicated chain of events? What are the systems that cause DNA to be methylated or not methylated, and how did we learn about them? Here is a good place to find interesting methylation-related phenomena and papers on them:

    > Bird, A. P. (1986). CpG-rich islands and the function of DNA methylation. Nature, 321(6067), 209-213. (**Old but good!**)
    
    >  Moore, L. D., Le, T., & Fan, G. (2013). DNA methylation and its basic function. Neuropsychopharmacology, 38(1), 23-38. https://www.nature.com/articles/npp2012112

- Transposable elements as a tumor suppressor system. Recent evidence builds towards a model where reduced DNA methylation in cancer cells leads to transposon expression and eventual cell death. What is the evidence for this model and how did it arise?

    > Zhao, Y., Oreskovic, E., Zhang, Q., Lu, Q., Gilman, A., Lin, Y. S., ... & Gorbunova, V. (2021). Transposon-triggered innate immune response confers cancer resistance to the blind mole rat. Nature Immunology, 1-12.

- Enhancers (fundamentals). Enhancers are regions of the genome that serve as "landing pads" to help transcriptional machinery activate nearby genes. There are many basic questions about enhancers that could serve as material for a project. For example: what was the first discovered enhancer and how was it found? What types of work have gone towards discovering enhancers in the human genome? If someone says there are 400,000 of them, or 1 million, where do those numbers come from? Are there different types of enhancers, and if so, how are they different? Are enhancers important in medical genetics, and if so, why do we think they are important? Pick one of these questions to start on. Useful starting points:

    > Pennacchio, L. A., Bickmore, W., Dean, A., Nobrega, M. A., & Bejerano, G. (2013). Enhancers: five essential questions. Nature Reviews Genetics, 14(4), 288-295.

    > Ray-Jones, H., Spivakov, M. Transcriptional enhancers and their communication with gene promoters. Cell. Mol. Life Sci. (2021). https://doi.org/10.1007/s00018-021-03903-w
    
- Enhancers (beta-interferon example). One particularly well-understood enhancer is beta-interferon. What happens to turn this gene on and off, and how do we know?

    > Munshi, N., Yie, J., Merika, M., Senger, K., Lomvardas, S., Agalioti, T., & Thanos, D. (1999, January). The IFN-β enhancer: a paradigm for understanding activation and repression of inducible gene expression. In Cold Spring Harbor symposia on quantitative biology (Vol. 64, pp. 149-160). Cold Spring Harbor Laboratory Press.

- Histone tail acetylation. In the late 1990's, it was known that transcription could be promoted by acetylation of a specific site on histone 3 (H3K27ac). Multiple explanations were drafted for why this might happen. What are some of those explanations and what have we since learned about them? A useful starting point:

    > Luger, K., & Richmond, T. J. (1998). The histone tails of the nucleosome. Current opinion in genetics & development, 8(2), 140-146.

- Nucleosomes (basic). We now know that DNA and nucleosomes are arranged in a pattern of "beads on a string" with DNA wrapped around one nucleosome after another. How do we know where the nucleosomes are and how they organize the DNA? 

    > Lai, W. K., & Pugh, B. F. (2017). Understanding nucleosome dynamics and their links to gene expression and DNA replication. Nature reviews Molecular cell biology, 18(9), 548-562.

- Nucleosomes (patterns). Nucleosomes display certain organized patterns or tendencies relative to the underlying DNA sequence. Specifically, they tend to sit with certain parts above bases G and C and others above bases A and T. Is this a simple, hard rule, or just a tendency among otherwise complicated or random behavior? What is the biological function of this type of arrangement? Be sure to convey the type of evidence that is used to answer those questions. 

    > Segal, E., Fondufe-Mittendorf, Y., Chen, L., Thåström, A., Field, Y., Moore, I. K., ... & Widom, J. (2006). A genomic code for nucleosome positioning. Nature, 442(7104), 772-778.
    
- Nucleosomes (transcription). Promoters are sites in genes where transcription starts. The DNA sequence and the proteins occupying the promoter are important in controlling whether the gene is transcribed. Part of this control is accomplished via nucleosome positioning: nucleosomes sit at regular intervals downstream of promoters and must sometimes be moved to turn on a gene. How do we know about these patterns and their biological functions?

    > Lai, W. K., & Pugh, B. F. (2017). Understanding nucleosome dynamics and their links to gene expression and DNA replication. Nature reviews Molecular cell biology, 18(9), 548-562.
    
    > Jiang, C., & Pugh, B. F. (2009). Nucleosome positioning and gene regulation: advances through genomics. Nature Reviews Genetics, 10(3), 161-172.

- Immune memory. Some very recent work argues that epigenetic changes help the immune system "remember" how to respond to a pathogen better after the second exposure than the first. How do we know that this memory exists and why do these authors believe it is epigenetic? 

    This mechanism is very different from the type of memory typically discussed when considering vaccines or prior infections. If you google "immune memory" you'll find the wrong concept. Instead, start with this paper and the ones it refers back to. 

    > de Laval, B., Maurizio, J., Kandalla, P. K., Brisou, G., Simonnet, L., Huber, C., ... & Sieweke, M. H. (2020). C/EBPβ-dependent epigenetic memory induces trained immunity in hematopoietic stem cells. Cell stem cell, 26(5), 657-674.

### Additional topics

You are welcome to propose additional topics. Here are some resources that might be useful.

- Interview and lab website from Job Dekker, an HHMI epigenetics researcher.

    > https://www.hhmi.org/news/packing-genome-step-by-step
    > http://www.dekkerlab.org/

- Interview and lab website from Oliver Rando, an epigenetics researcher at UMass Medical School.

    > http://symposium.cshlp.org/content/84/288.full
    > https://www.umassmed.edu/randolab/

- Interview and lab website from Andy Feinberg, an epigenetics researcher here at JHU. 

    > https://www.futuremedicine.com/doi/full/10.2217/epi.09.8
    > https://feinberglab.jhu.edu/index.php/what-is-epigenetics/
    